src/core/Database.cpp

```diff
@@ -159,6 +159,11 @@ bool Database::open(const QString& filePath, QSharedPointer<const CompositeKey>
     return true;
 }
 
+bool Database::isSaving()
+{
+    return !m_saveMutex.tryLock();
```

Comment:
What happens if tryLock() is successful? There should be a release call right after this statement.

Reply:
Correct this is an error. We may have to shift to a Boolean here.

Reply:
You could simply add m_saveMutex.unlock() afterwards.